common/rabbitmq/templates/headless-service.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if eq .Values.kind "statefulset" }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "fullname" . }}-pods
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+    type: rabbitmq
+    component: {{ .Release.Name }}
+    system: openstack
+spec:
+  selector:
+    app: {{ template "fullname" . }}
+  publishNotReadyAddresses: true  # See https://www.rabbitmq.com/cluster-formation.html#peer-discovery-k8s
```

Comment:
Where do you enable the `rabbitmq_peer_discovery_k8s` plugin or why would we need this besides that?

Reply:
I don't enable it, as it isn't supposed to be clustered setup. I think that is likely better solved with the rabbitmq operator.
It the proposal for a small iterative step consolidating the indepenedent set of rabbitmqs used for notifications in a statefulset with poddisruptionbudget.

I just set the head-less service up to make the dns records available like in the clustered setup. So that a consumer can access all the individual pods via dns and get the messages from there.
Getting the messages from the "normal" service wouldn't work as we would connect to a random host, having only a subset of the messages.

Reply:
My question aimed not at the headless service in general, but rather to that specific `publishNotReadyAddresses` option.
Why are you interested to be able to connect to pods that are not ready - is this something the consumers should be able to do on graceful shutdown or something?